Misc/NEWS.d/next/Library/2021-09-14-15-52-47.bpo-45192.DjA-BI.rst

```diff
@@ -0,0 +1,5 @@
+Fix the ``tempfile.__infer_return_type`` function so that the ``dir``
+argument of the :mod:`tempfile` functions accepts an object implementing the
+``os.PathLike[bytes]`` protocol.
```

Comment:
```suggestion
``os.PathLike`` protocol.
```

Reply:
`os.PathLike` is `os.PathLike[AnyStr]`.
I wanted to focus only on `os.PathLike[bytes]`, not `os.PathLike[str]`.
Do you think `os.PathLike` is better though?

Reply:
I think so, because [here](https://github.com/python/cpython/pull/28323/files#diff-6553a99f3ae04c9fc9c2349ac27037bfe3b274d5ca54d5800d9c5b5f11e29d21R92) you modify how all `os.PathLike` objects are handled, not just `PathLike[bytes]`

Reply:
You are right, thank you!